week2/Q2.py

```diff
@@ -0,0 +1,65 @@
+class BinaryTree:
```

Comment:
This is the same class as above. Put it instead in a different module and import it into your two answers to avoid copying it. Code that is copied somewhere else needs to be updated multiple times...